### PR TITLE
Add a few more arguments to allow errors during indexing

### PR DESCRIPTION
**Pull request by ahoppen**
I noticed that we were missing a bunch more arguments (in particular for clang index invocations), while diagnosing rdar://129071600, which is fixed by passing `-experimental-allow-module-with-compiler-errors` to the Swift index invocations.

rdar://129071600

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

File: Sources/SemanticIndex/UpdateIndexStoreTaskDescription.swift [hunk 1]
```diff
@@ -430,11 +430,6 @@ private func adjustSwiftCompilerArgumentsForIndexStoreUpdate(
     .option("output-file-map", [.singleDash], [.separatedBySpace, .separatedByEqualSign]),
   ]
 
-  let removeFrontendFlags = [
-    "-experimental-skip-non-inlinable-function-bodies",
-    "-experimental-skip-all-function-bodies",
-  ]
-
```
Comment (hamishknight) [inline Sources/SemanticIndex/UpdateIndexStoreTaskDescription.swift:437]: Don't we want to remove these for indexing?

File: Tests/SourceKitLSPTests/BackgroundIndexingTests.swift [hunk 2]
```diff
@@ -887,4 +887,47 @@ final class BackgroundIndexingTests: XCTestCase {
     _ = try project.openDocument("Lib.swift")
     _ = try await project.testClient.send(BarrierRequest())
   }
+
+  func testImportPreparedModuleWithFunctionBodiesSkipped() async throws {
+    // This test case was crashing the indexing compiler invocation for Client if Lib was built for index preparation
+    // (using `-enable-library-evolution -experimental-skip-all-function-bodies -experimental-lazy-typecheck`) but x
```
Comment (bnbarham) [inline Tests/SourceKitLSPTests/BackgroundIndexingTests.swift:893]: `x`?

File: Sources/SemanticIndex/UpdateIndexStoreTaskDescription.swift (hunk 1 above)
Comment (bnbarham) [reply Sources/SemanticIndex/UpdateIndexStoreTaskDescription.swift:437]: They would presumably only be there if we added them, which we're currently. not. Probably doesn't hurt to keep anyway though.

**Pull request merged**